applications/FluidDynamicsApplication/custom_elements/surface_smoothing_element.cpp

```diff
@@ -0,0 +1,525 @@
+                    const double face_weight = face_jacobians(i_gauss) * face_gauss_pts[i_gauss].Weight();
+
+                    // Compute the global coordinates of the face Gauss pt.
+                    global_coords = outer_face.GlobalCoordinates(global_coords, face_gauss_pts[i_gauss].Coordinates());
+
+                    // Compute the parent geometry local coordinates of the Gauss pt.
+                    loc_coords = geometry.PointLocalCoordinates(loc_coords, global_coords);
+
+                    // Compute shape function values
+                    // Obtain the parent subgeometry shape function values
+                    face_shape_func = geometry.ShapeFunctionsValues(face_shape_func, loc_coords);
+
+                    double temp_value = 0.0;
+
+                    for (unsigned int i = 0; i < num_nodes; i++){
+                        if (geometry[i].Is(CONTACT) )
+                        {
+                            solid_normal = geometry[i].FastGetSolutionStepValue(NORMAL);
+                            const double norm = Kratos::norm_2(solid_normal);
+#ifdef KRATOS_DEBUG
+                            KRATOS_WARNING_IF("DistanceSmoothingElement", norm < 1.0e-12) << "WARNING: Normal close to zero" <<std::endl;
+#endif
+                            solid_normal = (1.0/norm)*solid_normal;
+                        }
+
+                        temp_value += face_shape_func(i)*Kratos::inner_prod(solid_normal, grad_phi_old[i]);
+                    }
+
+                    for (unsigned int i = 0; i < num_nodes; i++){
+                        tempBCRHS[i] += epsilon * temp_value * face_weight * face_shape_func(i);
+                    }
+                }
+            }
+
+        }
+    }
+
+    noalias(rLeftHandSideMatrix) = tempM + tempA;
+    noalias(rRightHandSideVector) = tempBCRHS + prod(tempM,phi_old) - prod(rLeftHandSideMatrix,phi_dof);
+
+    KRATOS_CATCH("");
+}
+
+/**
+ * This method provides the place to perform checks on the completeness of the input
+ * and the compatibility with the problem options as well as the contitutive laws selected
+ * It is designed to be called only once (or anyway, not often) typically at the beginning
+ * of the calculations, so to verify that nothing is missing from the input
+ * or that no common error is found.
+ * @param rCurrentProcessInfo
+ * this method is: MANDATORY
+ */
+int SurfaceSmoothingElement::Check(const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_TRY
+
+    KRATOS_ERROR_IF(this->GetGeometry().Area() <= 0) << "On SurfaceSmoothingElement -> "
+        << this->Id() <<  "; Area cannot be less than or equal to 0" << std::endl;
+
+    // Base class checks for domain size and Id > 0
+    int ierr = Element::Check(rCurrentProcessInfo);
+    if(ierr != 0) return ierr;
+
+    // Check that all required variables have been registered
+    //KRATOS_CHECK_VARIABLE_KEY(DISTANCE)
+    //KRATOS_CHECK_VARIABLE_KEY(DISTANCE_GRADIENT)
```

Comment:
Remove commented code that is not required.

Reply:
Done